src/lib/seam/locks/use-toggle-lock-snackbar.tsx

```diff
@@ -0,0 +1,43 @@
+import { useCallback, useMemo, useState } from 'react'
+
+import { Snackbar, type SnackbarVariant } from 'lib/ui/Snackbar/Snackbar.js'
+
+export interface UseToggleLockSnackbarContext {
+  showToggleSnackbar: (variant: SnackbarVariant) => void
+  SnackbarNode: JSX.Element
+}
+
+export function useToggleLockSnackbar(): UseToggleLockSnackbarContext {
+  const [visible, setVisible] = useState(false)
+  const [variant, setVariant] = useState<SnackbarVariant>('success')
+
+  const SnackbarNode = useMemo(
+    () => (
+      <Snackbar
+        variant={variant}
+        visible={visible}
+        onClose={() => {
+          setVisible(false)
+        }}
+        message={
+          variant === 'success' ? t.successfullyUpdated : t.failedToUpdate
```

Comment:
Does this pattern mean we'll need a new hook (and file) for every single snackbar going forward? or maybe we only create one if it is being used in multiple places?

IMO duplicating seems simpler here instead of a separate hook.

Reply:
@razor-x  any thoughts on this?

Reply:
the reason why I created a hook is there is two different components that are toggling snackbars. In order not to duplicate the code I created a hook for it.

Reply:
Yea, I think we're ok to just duplicate it. Otherwise this pattern could result in too many hooks/files, for every snackbar type.

Reply:
@mikewuu updated it as using duplication